packages/babel-plugin-transform-async-to-promises/src/index.js

```diff
@@ -0,0 +1,112 @@
+/* Babel 7 transform */
+
+const babylon = require("babylon");
+const generate = require("@babel/generator").default;
+
+function printNode(ast) {
+  return generate(ast, {}).code;
+}
+
+function parse(code) {
+  return babylon.parse(code, {
+    allowImportExportEverywhere: true,
+    allowReturnOutsideFunction: true,
+    allowSuperOutsideMethod: true,
+  }).program;
+}
+
+const transform = require("nodent-transform").transform;
```

Comment:
This answers https://github.com/babel/babel/pull/7076/files#r160679424. Could we rather have a real Babel plugin?

Reply:
Not without a lot of effort - far more than I can commit to. This was discussed initially in https://github.com/MatAtBread/fast-async/issues/46 and at that point the costs of re-coding for the babel API (ignoring the stylistic requirements) would not be viable without the significant contribution of knowledgable babel contributors, hence a half-way where the 'transform' part of nodent was stripped of its dependancies was provided by me.